samples/snippets/src/main/java/com/example/bigquerystorage/WriteToDefaultStream.java

```diff
@@ -66,15 +69,73 @@ public static void writeToDefaultStream(String projectId, String datasetName, St
           record.put("test_string", String.format("record %03d-%03d", i, j));
           jsonArr.put(record);
         }
-        ApiFuture<AppendRowsResponse> future = writer.append(jsonArr);
-        AppendRowsResponse response = future.get();
+        writer.append(jsonArr);
       }
-      System.out.println("Appended records successfully.");
     } catch (ExecutionException e) {
       // If the wrapped exception is a StatusRuntimeException, check the state of the operation.
       // If the state is INTERNAL, CANCELLED, or ABORTED, you can retry. For more information, see:
       // https://grpc.github.io/grpc-java/javadoc/io/grpc/StatusRuntimeException.html
-      System.out.println("Failed to append records. \n" + e.toString());
+      System.out.println("Failed to append records. \n" + e);
+    }
+
+    // Final cleanup.
+    writer.cleanup();
+    System.out.println("Appended records successfully.");
+  }
+
+  public static WriteToDefaultStream create(String projectId, String datasetName, String tableName)
+      throws DescriptorValidationException, IOException, InterruptedException {
+    BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+    Table table = bigquery.getTable(datasetName, tableName);
+    TableName parentTable = TableName.of(projectId, datasetName, tableName);
+    Schema schema = table.getDefinition().getSchema();
+    TableSchema tableSchema = BqToBqStorageSchemaConverter.convertTableSchema(schema);
+
+    WriteToDefaultStream writer = new WriteToDefaultStream();
+    // Use the JSON stream writer to send records in JSON format.
+    // For more information about JsonStreamWriter, see:
+    // https://googleapis.dev/java/google-cloud-bigquerystorage/latest/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.html
+    writer.streamWriter = JsonStreamWriter.newBuilder(parentTable.toString(), tableSchema).build();
+    return writer;
+  }
+
+  public void append(JSONArray jsonArr)
+      throws DescriptorValidationException, IOException, ExecutionException {
+    ApiFuture<AppendRowsResponse> future = this.streamWriter.append(jsonArr);
+    this.inflightRequestPhaser.register();
+    ApiFutures.addCallback(
```

Comment:
Can we add a comment here to explain this?

Reply:
Added